API to import/export menu

evamillan opened the pull request:
- Unifies the routes to import and export the menu data into a single one with a GET and a PUT method:  `/_plugins/_bap/metadashboard`.
- Updates the README.
- Adds tests for the API. The tests can be run with `yarn test:api`.
- Adds a GitHub action that runs the tests.

sduenas commented on README.md:
````diff
@@ -77,6 +77,54 @@ folder the following command:
+Returns the menu data in a JSON format.
+
+```
+ GET /_plugins/_bap/metadashboard
````
You can add an example with `curl` to run the command.

sduenas commented on babel.config.js:
```diff
@@ -14,15 +15,15 @@
-    require('@babel/plugin-proposal-object-rest-spread'),
-  ],
-};
+ module.exports = {
+   presets: [
+     require('@babel/preset-env'),
+     require('@babel/preset-react'),
+     require('@babel/preset-typescript'),
+   ],
+   plugins: [
+     [require('@babel/plugin-transform-runtime'), { regenerator: true }],
+     require('@babel/plugin-proposal-class-properties'),
+     require('@babel/plugin-proposal-object-rest-spread'),
+   ],
+ };
```
Initial white spaces should be removed.

sduenas commented on public/components/dragDropEditor.tsx:
```diff
@@ -59,7 +60,7 @@ export const DragDropEditor = ({ value, http, renderToast }) => {
     try {
       const body = JSON.stringify({ metadashboard });
-      const response = await http.put('/api/dashboards/metadashboard/edit', {
+      const response = await http.put(`${API_PREFIX}/metadashboard`, {
```
A better name would be `menu` because we only add data for the bap menu and not something else.

sduenas reviewed (changes requested):
LGTM. It works as expected but I added some minor comments.

evamillan commented:
@sduenas I've added the example, removed the white spaces and renamed all references from "metadashboard" to "menu". This means that all menus will need to be updated once this is in production.

sduenas reviewed (approved):
LGTM
